test/hotspot/jtreg/runtime/ErrorHandling/HsErrFileUtils.java

```diff
@@ -132,7 +132,14 @@ public static void checkHsErrFileContent(File f, Pattern[] positivePatterns, Pat
             }
             // Found all positive pattern?
             if (!positivePatternStack.isEmpty()) {
-                throw new RuntimeException("hs-err file incomplete (first missing pattern: " + positivePatternStack.peek() + ")");
+                if (currentPositivePattern != null) {
+                    throw new RuntimeException("hs-err file incomplete (first missing pattern: " + currentPositivePattern.pattern() + ")");
+                } else {
+                    // Impossible to reach here, where stack is not empty and currentPositivePattern which was peeked from stack is null.
```

Comment:
If this is impossible to reach then why put in the null check? If the impossible were to happen you would get a NPE.

Reply:
`currentPositivePattern` is null checked before any usage. Therefore, the other parts of the code are guarded against NPE.
For using `currentPositivePattern`, I should also check it against `null`. Then, the `else { ...} ` part became unreachable. I just put this part for future contributors. Maybe the comments only are enough for that.
What is better: remove else part? or improve the comments? or both?

Reply:
IMO all you need here is simply:
```
// Found all positive pattern?
if (!positivePatternStack.isEmpty()) {
    throw new RuntimeException("hs-err file incomplete (first missing pattern: " + currentPositivePattern.pattern() + ")");
}
```